ios/FluentUI/Badge Field/BadgeView.swift

```diff
@@ -187,11 +137,148 @@ open class BadgeView: UIView {
 
     @objc open var isSelected: Bool = false {
         didSet {
-            updateBackgroundColor()
-            updateLabelTextColor()
+            updateColors()
             updateAccessibility()
         }
     }
+    
+    private var _labelTextColor: UIColor?
+    open var labelTextColor: UIColor? {
+        get {
+            if let customLabelTextColor = _labelTextColor {
+                return customLabelTextColor
+            }
+            
+            switch style {
+                case .default:
+                    if let window = window {
+                        return Colors.primary(for: window)
+                    }
+                case .warning:
+                    return Colors.Badge.textWarning
+                case .error:
+                    return Colors.Badge.textError
+            }
+            return nil
```

Comment:
is this last return nil needed?
the selectedLabelTextColor below does not have the same logic.

Reply:
This is mostly because other colors are have a case where in default style the view isn't added in the container yet. 
In this case window would be nil and color is undefined. 

`selectedLabelTextColor` doesn't have any colors dependent on window property.